src/SelfRoleManagerEvents.ts

```diff
@@ -14,6 +14,7 @@ export enum SelfRoleManagerEvents {
   interaction = 'interaction',
 
   maxRolesReach = 'maxRolesReach',
+  notMetDependencies = 'notMetDependencies',
```

Comment:
Consistency obliges... if you remove the "dependencies" name, you should align yourself everywhere. This event could be named `conditionsNotMet` or `requiredRolesMissing` for example.